components/plan-build-ps1/bin/hab-plan-build.ps1

```diff
@@ -1138,6 +1138,19 @@ function __resolve_all_version_placeholders($env_table, $real_version) {
     }
 }
 
+# Replace all instances of the "__pkg__version__unset__" placeholder
+# in the values of the given hashtable with the real version number.
+function __resolve_all_version_placeholders_for_provenance($provenance_table, $real_version) {
+  $new_table = @{}
+  foreach($k in $provenance_table.keys) {
+      $new_table[$k] = (__resolve_version_placeholder $provenance_table[$k] $real_version)
```

Comment:
Per the discovery in #6992, we should explicitly document significance of the `.Value` accessor (or lack thereof) for both this and the "environment" function.

Reply:
added those comments